pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -318,6 +318,12 @@ func (r *HypershiftDeploymentReconciler) destroyHypershift(hyd *hypdeployment.Hy
 
 	inHyd := hyd.DeepCopy()
 
+	// if the hostedcluster has a managed cluster, wait for its managed cluster to be cleaned up
+	if controllerutil.ContainsFinalizer(hyd, constant.ManagedClusterCleanupFinalizer) {
+		log.Info("Waiting for ManagedCluster " + helper.ManagedClusterName(hyd) + " to be cleaned up, retry in 10s")
+		return ctrl.Result{RequeueAfter: 10 * time.Second, Requeue: true}, nil
```

Comment:
I think we don't need the requeue here since removing the `ManagedClusterCleanupFinalizer` will trigger an update event.

Reply:
yes, you're right, updated